README_developers.md

```diff
@@ -91,128 +85,356 @@ $ # For a fix
+- Comma separator `,` should have space after, but not before.
 
-// Bad
-// this causes confusion when working with pointers, and if you
-// must change a type in the future, we will have a larger diff.
-int a, b, c;
+### Pointers and references
 
-// Good
-// uses allman braces
-// no space between if and ()
-if(some_condition)
-{
+The character `*` is a part of the pointer type name and there
+should be no space between the type name and `*`. Similar for
+the character `&` in reference types.
+
+```c++
+int* pint;      // Yes!
+double& refdbl; // Yes!
+
+int *pint;       // No!
+double & refdbl; // No!
+```
+### Flow control statements
+
+No spaces between `if` (and other control/loop statements) and `(`.
+However, do make space before opening brace `{`.
+```c++
+// Preferred: no space between if and ()
+if(some_condition) {
   value += 1;
 }
 
-// Not preferred
-// use allman braces
-if(some_condition) {
+// Discouraged: space in between if and ()
+if (some_condition) {
   value += 1;
 }
 
-// Not preferred
-// space in between if and ()
-if (some_condition)
+// Discouraged: no space between ) and {.
+for(some_condition){
+  value += 1;
+}
+```
+
+## Braces
+
+### Use K&R style
+
+Have the opening brace on the same line as the
+statement and the closing brace aligned with the statement.
+```c++
+// Yes: Use K&R braces
+while(a > 0) {
+  value += 1;
+}
+
+// No: Allman braces
+while(a > 0)
 {
   value += 1;
 }
 
-// Not preferred
-// please use braces
-if(some_condition)
+// No: GNU braces
+while(a > 0)
+  {
+    value += 1;
+  }
+```
+
+### Position of `else` keyword
+
+Keyword `else` should follow the if closing brace. 
+
+```c++
+// Yes: else keyword follows `if` closing brace
+if(a > 0) {
   value += 1;
+  b = a;
+} else {
+  value -= 1;
+}
+
+// No: else on the next line
+if(a > 0) {
+  value += 1;
+  b = a;
+}
+else {
+  value -= 1;
+}
+```
+### Use braces for every block
+
+For consistency and better code readability use braces
+for every code block in selection statements. Also,
+avoid condition and loop bodies in the same line of code
+```c++
+// Yes: Use braces for each block
+if(a > 0) {
+  value += 1;
+  b = a;
+} else {
+  value -= 1;
+}
+
+// No: Use braces only for blocks with more than one statement 
+if(a > 0) {
+  value += 1;
+  b = a;
+} else
+  value -= 1;
+
+// No: the condition and the body on the same line 
+if (a > 0) value += 1;
+else       value -= 1;
+```
 
-// Bad!
-if(some_condition) value += 1;
 
+### Do not use one true brace style
+
+For function and class definitions, have the opening brace
+on the next line aligned with the definition statement.
+```c++
+// Yes: opening brace on the next line for functions and classes declarations
+bool MyClass::my_method(int a)
+{
+
+}
+
+// No: opening brace on the same line
+bool MyClass::my_method(int a) {
+
+}
+``` 
+
+## RAJA specific style
+- Indent two spaces above surrounding scope
+- No spaces around template parameter
+- Line break before range segment
```

Comment:
Every policy in the document besides this one would be relatively easy to automate - if you're sure about this one we can leave it but tools may have a harder time enforcing.

Reply:
you mean "Line break before range segment" ?